File: Programmers/c++/문자열 내 마음대로 정렬하기.cpp
/*
프로그래머스 문자열 내 마음대로 정렬하기
vector에 페어형태로 n번째 문자와 본 단어를 저장
sort로 정렬
answer에 vector에 넣은 단어를 순서대로 삽입
*/
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

vector<string> solution(vector<string> strings, int n) {
	vector<string> answer;
	vector<pair<char, string>> v;
	for (int i = 0; i<strings.size(); i++) {
		v.push_back(make_pair(strings[i][n], strings[i]));
	}
	sort(v.begin(), v.end());
	for (int i = 0; i<v.size(); i++) {
		answer.push_back(v[i].second);
	}
	return answer;
}

File: Programmers/javascript/문자열 내 마음대로 정렬하기.js
/*
프로그래머스 문자열 내 마음대로 정렬하기
주어진 문자열을 2가지 기준으로 비교하여 sort
1. n번째 문자로 정렬
2. n번째 문자가 같다면 사전순서로 정렬
*/
function solution(strings, n) {
    var answer = strings;
    
    answer.sort(function(a,b){
        if(a[n]<b[n]) return -1;
        else if(a[n]==b[n]) {
            if(a<b) return -1;
        }
    });
    
    return answer;
}
